### PR TITLE
docs: adding docstrings for the SparseAEPipeline class

### LINKED ISSUE #59: Improve SparseAEPipeline docs
Add docstrings to the SparseAEPipeline. It follows the paper: https://web.stanford.edu/class/cs294a/sparseAutoencoder.pdf

### PR DESCRIPTION
**Pull request by REETI-SARUP**
Fixes #59 

Signed-off-by: Reeti Sarup <reeti.sarup@gmail.com>

**Explain what this PR does:**
Adding docstrings for the `SparseAEPipeline` class & its methods in the `models/autoencoder/pipeline.py` file.

### CODE REVIEW

File: numalogic/models/autoencoder/pipeline.py
Comment (ab93) [inline numalogic/models/autoencoder/pipeline.py:383]: You can remove the docstring in __init__, since you already added that in the class docstring.

**Review by ab93 — CHANGES_REQUESTED** [review]
Thanks @REETI-SARUP. A minor comment. Also, can you run `make lint` locally, and push the changes? The linting check is failing.

File: numalogic/models/autoencoder/pipeline.py
Comment (REETI-SARUP) [reply numalogic/models/autoencoder/pipeline.py]: Sure, done.

**Comment by REETI-SARUP** [comment]
> Thanks @REETI-SARUP. A minor comment. Also, can you run `make lint` locally, and push the changes? The linting check is failing.

Sure @ab93, thanks, I've addressed the review suggestion and pushed the linting fixes for `pipeline.py`. Could you please re-run the workflows / checks again on the PR to see if the linting check is passing now in the PR, it is currently showing that running the workflows / checks on the PR requires approval from maintainer of this repo.

**Review by ab93 — APPROVED** [review]
Looks great @REETI-SARUP ! Merging.

**Pull request merged**
